src/plugins/guided_onboarding/server/routes/guide_state.routes.ts

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { IRouter, SavedObjectsClient } from '@kbn/core/server';
+import { schema } from '@kbn/config-schema';
+import { GuideId, GuideState } from '@kbn/guided-onboarding';
+import {
+  getDefaultGuideState,
+  updateGuideState,
+  findGuideById,
+} from '../helpers/guide_state.utils';
+import { API_BASE_PATH, GuidesConfig } from '../../common';
+import { findAllGuides, getPluginState, updatePluginStatus } from '../helpers';
+
+/*
+ * This route is only used in the example plugin.
+ */
+export const registerGetGuideStateRoute = (router: IRouter) => {
+  // Fetch all guides state
+  router.get(
+    {
+      path: `${API_BASE_PATH}/guides`,
+      validate: false,
+    },
+    async (context, request, response) => {
+      const coreContext = await context.core;
+      const soClient = coreContext.savedObjects.client as SavedObjectsClient;
+
+      const existingGuides = await findAllGuides(soClient);
+
+      if (existingGuides.total > 0) {
+        const guidesState = existingGuides.saved_objects.map((guide) => guide.attributes);
+        return response.ok({
+          body: { state: guidesState },
+        });
+      } else {
+        // If no SO exists, we assume state hasn't been stored yet and return an empty array
+        return response.ok({
+          body: { state: [] },
+        });
+      }
+    }
+  );
+};
+
+export const registerActivateGuideRoute = (router: IRouter, guidesConfig: GuidesConfig) => {
```

Comment:
This file was renamed and the only change is addition of the "activate guide" route